app/src/main/java/com/gh4a/activities/PullRequestActivity.java

```diff
@@ -614,4 +676,56 @@ protected String getErrorMessage() {
             return getContext().getString(R.string.pull_error_merge, mPullRequest.getNumber());
         }
     }
+
+
+    private class PullRequestReviewTask extends ProgressDialogTask<Review> {
+        private final String mReviewMessage;
+        private final String mReviewEvent;
+
+        public PullRequestReviewTask(String reviewMessage, String reviewEvent) {
+            super(getBaseActivity(), R.string.reviewing_msg);
+            mReviewMessage = reviewMessage;
+            mReviewEvent = reviewEvent;
+        }
+
+        @Override
+        protected ProgressDialogTask<Review> clone() {
+            return new PullRequestReviewTask(mReviewMessage, mReviewEvent);
+        }
+
+        @Override
+        protected Review run() throws Exception {
+            PullRequestService pullService = (PullRequestService)
+                    Gh4Application.get().getService(Gh4Application.PULL_SERVICE);
+            RepositoryId repoId = new RepositoryId(mRepoOwner, mRepoName);
+            return pullService.createPullRequestReview(repoId, mPullRequest.getNumber(), mReviewEvent, mReviewMessage);
+        }
+
+        @Override
+        protected void onSuccess(Review result) {
+        }
+
+        @Override
+        protected String getErrorMessage() {
+            /* How do you get at the response/result of pullService.createPullRequestReview ?? */
```

Comment:
No, there  isn't. If it was available, there would be no reason to show an error message ;-)

Just to make it clear: the return value of getErrorMessage is shown in a Snackbar if the task failed (e.g. due to network issues)

Reply:
I think my mental model is suffering here.. 

in my native language ( python ), i'd expect to be able to get to the result of the call to the github api without regard to the success or failure of the call. 

In my mind, I expect `result` to be extant if the call succeeded or failed ( or accessible in some way ) 

example below with the python REPL and a module named requests. This call `401`ed. 

```text
>>> req = requests.post("https://api.github.com/repos/slapperwan/gh4a/pulls/790/reviews", headers=auth, json={'event':'COMMENT','comments':'example response'})
>>> req.status_code
401
>>> req.json()
{u'documentation_url': u'https://developer.github.com/v3', u'message': u'Requires authentication'}
```

Reply:
i want to say something like the below, but it doesn't seem to be ideomatic java

```java
@Override
protected String getErrorMessage(result) {
            return result.response_body['message'];
}
```

Reply:
> example below with the python REPL and a module named requests.

And where is the review object in that response? ;-)
Exactly that is the point: the bindings here don't return the raw response, but unpack the JSON to a Java object on success and throw an exception on failure, with success being defined as a 20x HTTP status code. If there's an error response, there's no JSON to be interpreted.

Reply:
> i want to say something like the below, but it doesn't seem to be ideomatic java

Right, and now I understand. That's not how we usually do the Snackbars though. A text like 'Review creation failed.' is totally sufficient.

Reply:
ah, that makes sense. I was thinking about it from the perspective of having access to the raw http responses, not the Objectified review response. 👍👍